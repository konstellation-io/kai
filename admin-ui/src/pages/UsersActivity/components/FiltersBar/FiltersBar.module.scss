@import '../../../../styles/mixins';
@import '../../../../styles/colors';

.formField {
  min-height: 140px;
  height: auto;
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  margin-top: 30px;
  border: solid 2px $bg-color-lighter;
  border-width: 2px 0;

  .label {
    @include font-body;
    color: font-color(light);

    margin: 0 40px 0 30px;
  }

  > div:not(:last-child) {
    flex: 1 1 0px;
    min-width: 210px;
    margin-right: 2vw;
  }

  svg {
    margin-left: 20px;
  }
}

.buttons {
  display: flex;
  flex-direction: row;
}
