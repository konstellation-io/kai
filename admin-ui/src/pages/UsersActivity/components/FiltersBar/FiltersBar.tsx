import React from 'react';

import Calendar from '../../../../components/Form/Calendar/Calendar';
import Select from '../../../../components/Form/Select/Select';
import Button from '../../../../components/Button/Button';
import { Form } from '../../../../hooks/useForm';

import styles from './FiltersBar.module.scss';
import { ApolloError } from 'apollo-client';

const typeToText = {
  LOGIN: 'Login',
  LOGOUT: 'Logout',
  CREATE_RUNTIME: 'Runtime Creation'
};

type FormFieldProps = {
  error?: ApolloError;
  form: Form;
  onSubmit: Function;
  types: string[];
  users: string[];
};
function FiltersBar({ form, onSubmit, types, users }: FormFieldProps) {
  return (
    <div className={styles.formField}>
      <Select
        options={types}
        onChange={form.input.type.onChange}
        error={form.input.type.error}
        formSelectedOption={form.input.type.value}
        label="Activity type"
        placeholder="Activity type"
        valuesMapper={typeToText}
      />
      <Select
        options={users}
        onChange={form.input.userEmail.onChange}
        error={form.input.userEmail.error}
        formSelectedOption={form.input.userEmail.value}
        label="User email"
        placeholder="User email"
      />
      <Calendar
        onChangeFromDate={form.input.fromDate.onChange}
        onChangeToDate={form.input.toDate.onChange}
        formFromDate={form.input.fromDate.value}
        formToDate={form.input.toDate.value}
        error={form.input.fromDate.error || form.input.toDate.error}
      />
      <div className={styles.buttons}>
        <Button
          label={'SEARCH'}
          onClick={onSubmit}
          border
          style={{ margin: '24px 0' }}
        />
        <Button
          label={'CLEAR'}
          onClick={() => {
            form.clearInputs(true);
          }}
          style={{ margin: '24px 0' }}
        />
      </div>
    </div>
  );
}

export default FiltersBar;
