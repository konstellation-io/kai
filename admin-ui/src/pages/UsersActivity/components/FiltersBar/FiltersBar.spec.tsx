import React from 'react';
import { shallow } from 'enzyme';
import FiltersBar from './FiltersBar';

describe('FiltersBar', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <FiltersBar
        onSubmit={jest.fn()}
        types={['a', 'b', 'c']}
        users={['1', '2', '3']}
      />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
