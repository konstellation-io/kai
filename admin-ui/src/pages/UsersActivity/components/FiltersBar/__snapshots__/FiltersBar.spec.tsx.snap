// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersBar matches snapshot 1`] = `
<div
  className="formField"
>
  <Select
    label="Activity type"
    onChange={[MockFunction]}
    options={
      Array [
        "a",
        "b",
        "c",
      ]
    }
    placeholder="Activity type"
    valuesMapper={
      Object {
        "CREATE_RUNTIME": "Create Runtime",
        "CREATE_VERSION": "Create Version",
        "LOGIN": "Login",
        "LOGOUT": "Logout",
        "PUBLISH_VERSION": "Publish Version",
        "START_VERSION": "Start Version",
        "STOP_VERSION": "Stop Version",
        "UNPUBLISH_VERSION": "Unpublish Version",
        "UPDATE_SETTING": "Update Settings",
        "UPDATE_VERSION_CONFIGURATION": "Update Version Configuration",
      }
    }
  />
  <Select
    label="User email"
    onChange={[MockFunction]}
    options={
      Array [
        "1",
        "2",
        "3",
      ]
    }
    placeholder="User email"
  />
  <Calendar
    onChangeFromDate={[MockFunction]}
    onChangeToDate={[MockFunction]}
  />
  <div
    className="buttons"
  >
    <Button
      border={true}
      label="SEARCH"
      onClick={[MockFunction]}
      style={
        Object {
          "margin": "24px 0",
        }
      }
    />
    <Button
      label="CLEAR"
      onClick={[Function]}
      style={
        Object {
          "margin": "24px 0",
        }
      }
    />
  </div>
</div>
`;
